lib/queue.rb

```diff
@@ -1,28 +1,33 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = []
+    @first = 0
+    @last = 0
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    @store[@last] = element
+    @last = (@last + 1) % 20
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    value = @store[@first]
+    @store.delete_at(@first)
+    @last = (@last - 1) % 20
+    return value
```

Comment:
Note that `delete_at` is an O(n) operation and shortens the array.  Better to do this:

```suggestion
    value = @store[@first]
    @store[@first] = nil
    @last = (@last - 1) % @store.length
    return value
```